runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -6698,6 +6698,149 @@ bool TR_EscapeAnalysis::tryToZeroInitializeUsingArrayset(Candidate* candidate, T
    return false;
    }
 
+void TR_EscapeAnalysis::makeNewValueLocalAllocation(Candidate *candidate)
+   {
+   //
+   // Example of trees for a newvalue operation before stack allocation
+   //
+   //    112n     treetop
+   //    n111n       newvalue  jitNewValue
+   //    n108n         loadaddr  Pair
+   //    n109n         iload x
+   //    n119n         iconst 2 (X!=0 X>=0 )
+   //
+   // and after, for contiguous stack allocation, where #439 is aggregate storage on the stack.
+   // The field holding the vft for the object will be inserted in the method entry block.
```

Comment:
My understanding is that it's because GC considers stack allocated objects in the process of garbage collection.  Stack allocated objects are considered to be live throughout the method, so the vft needs to be stored in object at method entry, and any reference fields in the object need to be zero initialized at method entry as well, so GC handles the objects (and anything they reference) correctly.

Reply:
My understanding is that it's because GC considers stack allocated objects in the process of garbage collection. Stack allocated objects are considered to be live throughout the method, so the vft needs to be stored in the object header at method entry, and any reference fields in the object need to be zero initialized at method entry as well, so GC handles the objects (and anything they reference) correctly.

Reply:
Thank you for the clarification Henry!